mono/utils/mono-error-internals.h

```diff
@@ -126,7 +126,25 @@ do { 							\
 void
 mono_error_assert_ok_pos (MonoError *error, const char* filename, int lineno) MONO_LLVM_INTERNAL;
 
-#define mono_error_assert_ok(e) mono_error_assert_ok_pos (e, __FILE__, __LINE__)
+/*
+ * Three macros to assert that a MonoError is ok:
+ * 1. mono_error_assert_ok(e) when you just want to print the error's message on failure
+ * 2. mono_error_assert_ok(e,msg) when you want to print "msg, due to <e's message>"
+ * 3. mono_error_assertf_ok(e,fmt,args...) when you want to print "<formatted msg>, due to <e's message>"
+ *    (fmt should specify the formatting just for args).
+ *
+ * What's the difference between mono_error_assert_msg_ok (e, "foo") and
+ * mono_error_assertf_ok (e, "foo") ?  The former works as you expect, the
+ * latter unhelpfully expands to
+ *
+ * g_assertf (is_ok (e), "foo, due to %s", ,  mono_error_get_message (err)).
+ *
+ * Note the double commas.  Turns out that to get rid of that extra comma
+ * portably we would have to write really ugly preprocessor macros.
+ */
+#define mono_error_assert_ok(err)          g_assertf (is_ok (err), "%s", mono_error_get_message (err))
+#define mono_error_assert_msg_ok(err,msg)  g_assertf (is_ok (err), msg ", due to %s", mono_error_get_message (err))
```

Comment:
Spaces after commas please imho.

Reply:
Do we want to call this error instead of err for search purposes, i.e. find mostly all MonoError occurences?